src/composables/accounts.ts

```diff
@@ -26,11 +27,30 @@ export function useAccounts({ store }: IDefaultComposableOptions) {
     return accountsAddressList.value.includes(address);
   }
 
+  /**
+   * This function is used to format the given address.
+   * @param {string} address - the address to format.
+   * @returns {string} - returns the account's name if it exists,
+   * otherwise it returns the translated string "Account" + the account's index +1.
+  */
+  function getAccountNameByAddress(address: string): string {
+    const _account = accounts.value.find(
+      (acc: IAccount) => acc.address === address,
+    );
+
+    if (!_account?.idx) return address;
+
+    if (_account.name) return _account.name;
+
+    return `${i18n.t('pages.account.heading')} ${_account.idx + 1}`;
```

Comment:
If you test if the `idx` exists the return should have something to do with the idx. Now the if statements are messed up between each other.

Maybe this would be better:
```
    if (name) {
      return name;
    }
    if (idx) {
      return `${i18n.t('pages.account.heading')} ${idx + 1}`;
    }
    return address;
```